frontend/apps/remark42/app/common/fetcher.test.ts

```diff
@@ -125,6 +150,133 @@ describe('fetcher', () => {
+    });
+
+    it('should call setAuthCookie with proper parameters when receiving JWT token', async () => {
+      // Spy on setAuthCookie calls
+      const setAuthCookieSpy = jest.spyOn(cookies, 'setAuthCookie').mockImplementation(() => {});
+
+      // Create test JWT token
+      const jwtToken = 'test.jwt.token';
+
+      // Mock parseJwtPayload implementation since it's not directly accessible
+      jest.spyOn(window, 'atob').mockImplementation(() => JSON.stringify({ jti: 'test-jti-id', sub: '1234567890' }));
+
+      mockFetch({ headers: { [JWT_HEADER]: jwtToken, ...headers } });
+      await apiFetcher.get(apiUri);
+
+      // Verify setAuthCookie was called with expected parameters
+      expect(setAuthCookieSpy).toHaveBeenCalledWith(
+        JWT_COOKIE_NAME,
+        jwtToken,
+        expect.objectContaining({ expires: AUTH_COOKIE_TTL_SECONDS })
+      );
+
+      expect(setAuthCookieSpy).toHaveBeenCalledWith(
+        XSRF_COOKIE,
+        'test-jti-id',
+        expect.objectContaining({ expires: AUTH_COOKIE_TTL_SECONDS })
+      );
+
+      // Clean up
+      jest.restoreAllMocks();
+    });
+
+    it('should handle errors when setting cookies', async () => {
+      // Mock console.error
+      const originalConsoleError = console.error;
+      console.error = jest.fn();
+
+      // Make setAuthCookie throw an error
+      jest.spyOn(cookies, 'setAuthCookie').mockImplementation(() => {
+        throw new Error('Cookie access denied');
+      });
+
+      // Create test JWT token
+      const jwtToken = 'test.jwt.token';
+
+      // Mock parseJwtPayload implementation since it's not directly accessible
+      jest.spyOn(window, 'atob').mockImplementation(() => JSON.stringify({ jti: 'test-jti-id', sub: '1234567890' }));
+
+      mockFetch({ headers: { [JWT_HEADER]: jwtToken, ...headers } });
+
+      // This should not throw despite cookie setting failing
+      await apiFetcher.get(apiUri);
+
+      // Error should be logged
+      expect(console.error).toHaveBeenCalled();
+
+      // Restore mocks
+      console.error = originalConsoleError;
+      jest.restoreAllMocks();
+    });
+
+    it('should reset activeJwtToken and clear cookies on 401/403 responses', async () => {
+      // Mock clearAuthCookie
+      const clearAuthCookieSpy = jest.spyOn(cookies, 'clearAuthCookie').mockImplementation(() => {});
+
+      // Setup JWT token with mocked payload
+      const jwtToken = 'test.jwt.token';
+
+      // Mock JWT parsing
+      const mockPayload = { jti: 'test-jti-id', sub: '1234567890' };
+      jest.spyOn(window, 'atob').mockImplementation(() => JSON.stringify(mockPayload));
+
+      // First set JWT token
+      mockFetch({ headers: { [JWT_HEADER]: jwtToken, ...headers } });
+      await apiFetcher.get(apiUri);
+
+      // Now trigger a 401 response
+      mockFetch({ status: 401 });
+
+      try {
+        await apiFetcher.get(apiUri);
+      } catch (e) {
+        // Expected to fail with 401
+      }
```

Comment:
on the first test run snapshot of the error is going to be generated
```suggestion
       expect(() => apiFetcher.get(apiUri)). toThrowErrorMatchingInlineSnapshot();
```